src/main/java/io/confluent/connect/jdbc/sink/JdbcDbWriter.java

```diff
@@ -25,35 +25,29 @@
 import java.util.Map;
 
 import io.confluent.connect.jdbc.dialect.DatabaseDialect;
-import io.confluent.connect.jdbc.util.CachedConnectionProvider;
 import io.confluent.connect.jdbc.util.TableId;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import javax.sql.DataSource;
+
 public class JdbcDbWriter {
   private static final Logger log = LoggerFactory.getLogger(JdbcDbWriter.class);
 
   private final JdbcSinkConfig config;
   private final DatabaseDialect dbDialect;
   private final DbStructure dbStructure;
-  final CachedConnectionProvider cachedConnectionProvider;
```

Comment:
Look at the possiblity of removing or deprecating this class from the codebase both from sink and source connector. Other than the task.class, writer.class it will have some usage in the SourceConnector.class.